Closes-Bug: issues: #12: Backup/Restore failing for Azure due to VM cid change

pritishmishra opened the pull request:
Description: Using agent_id instead of cid for instance_id during backup/restore for Azure

pritishmishra commented:
This change has been tested and backup/restore functionalities are working fine.

jagadish-kb reviewed (approved):
LGTM

kamath-prasad commented on lib/agent/backuprestore.js:
```diff
@@ -49,9 +49,8 @@ function startBackup(params) {
     .filter(vm => _.eq(vm.job, job.name) && _.eq(vm.index, job.index))
     .first()
-    .get('cid')
+    .get(iaasCredentials.name === 'azure' ? 'agent_id' : 'cid')
```
please update to use new structure of input data here and below.

Commit:
Closes-Bug: issues: #12: Backup/Restore failing for Azure due to VM cid change
Description: Using new param, 'iaas_vm_metadata.vm_id', instead of cid for instance_id during backup/restore for all IaaS

pritishmishra commented:
Modified as per the new changes from the broker's side. Kindly review again.

kamath-prasad reviewed (approved):
LGTM . merging.
